src/variable_info.hpp

```diff
@@ -16,98 +16,119 @@
 
 #pragma once
 
-/** Information on a WML variable. */
-#include <string>
 #include "config.hpp"
 #include "variable_info_detail.hpp"
 
-class invalid_variablename_exception : public std::exception
+#include <string>
+#include <type_traits>
+
+// Restrict alias to local scope.
+namespace
```

Comment:
That would work if this was a source file, but it's not. Anonymous namespaces (or file-statics) make something local to the _translation unit_, not the file. That means it's local to the cpp file that (directly or indirectly) includes this file. Since there are presumably multiple such files, that generally means that each of those files gets their own instance of it. Fortunately, the only thing you've put here in this case is a namespace alias, which doesn't have linkage, but if it was a class, there could be problems - for example, if you declared `invalid_variablename_exception` in an anonymous namespace in a header, catch clauses for that exception likely wouldn't work since they end up referring to different types that just happen to have the same name.

If you want the namespace alias to be localized, I'd put it in the class itself, if possible (though I'm not sure if that's allowed). Another possibility would be a `using namespace` in the class (I believe that's allowed).